### PR TITLE
Make output channel names consistent

### PR DESCRIPTION
**Pull request by dhruvisompura**
### Description
Addresses #1707

@:output

We've received feedback that the output channel list is confusing and not easy to parse. A lot of the channel names are prefixed with "Positron" which is redundant. The formatting of the names for the language channels is not consistent and adds friction to the UX when scanning the list.

The proposed change here does some tidying 🧹 of the names to improve the UX:
* The "Positron" prefix has been removed
* The language specific channel names have a consistent formatting which groups them together in the select
* Updates ambiguous output channel names

### Screenshots

Screenshot 2024-12-19 at 10 20 58 AM

Screenshot 2024-12-19 at 10 25 38 AM

### CODE REVIEW

**Comment by jthomasmock** [comment]
Love it! Dhruvi and I synced up briefly, while we'd love to maybe someday have nice grouping or other organization of the Positron components (such as consoles together, kernels, or language runtimes, etc), making the naming nice and consistent is a great first pass.

**Review by juliasilge — APPROVED** [review]
This is working great for me, and the naming updates are a HUGE improvement:

Screenshot 2024-12-19 at 4 32 26 PM

I agree with @jthomasmock that eventually we would like to have more organization here so it is easier for users to find the relevant channels as tracked in #5063, but this PR is a big step in the right direction in terms of clarity and consistency.

File: extensions/positron-python/src/client/common/utils/localize.ts [hunk 1]
```diff
@@ -266,7 +266,7 @@ export namespace InterpreterQuickPickList {
 
 export namespace OutputChannelNames {
     export const languageServer = l10n.t('Python Language Server');
-    export const python = l10n.t('Python');
+    export const python = l10n.t('Python Language Pack');
```
Comment (juliasilge) [inline extensions/positron-python/src/client/common/utils/localize.ts:269]: I believe we need our [overlays](https://connect.posit.it/positron-wiki/overlay-strategy.html) in this file, to help with upstream merges of this extension.

**Review by juliasilge — CHANGES_REQUESTED** [review]

File: extensions/positron-python/src/client/common/utils/localize.ts (hunk 1 above)
Comment (dhruvisompura) [reply extensions/positron-python/src/client/common/utils/localize.ts:269]: Oh, you are right! Thank you for catching that.

**Review by juliasilge — APPROVED** [review]
Good to go once everything is green! ✅

**Pull request merged**